server/rest/src/test/scala/org/infinispan/rest/IntegrationTest.scala

```diff
@@ -689,6 +689,8 @@ class IntegrationTest extends RestServerTestBase {
          eTagFirst.equals(eTagSecond))
    }
 
+   // This test is *DISABLED* because of ISPN-5992
+   @Test(enabled = false)
```

Comment:
The test is just fine, but the 'production' logic is broken, and this commit reveals the problem (the test crashes as it should crash).